### ISSUE TITLE
Version 6.1 with parallel_tests gem is not working anymore

### ISSUE THREAD

**Issue by Piioo** [cockroachdb/activerecord-cockroachdb-adapter#204]
Hello,

I tried to update the gem to the stable 6.1 version and our application is working fine, but we have problems now with the parallel_tests gem.
Im not sure, but it cant create databases anymore.

```
➜  foo git:(chore/update_activerecord-cockroachdb-adapter_non_beta) bundle exec rake parallel:setup
rake aborted!
ActiveRecord::NoDatabaseError: PG::InvalidCatalogName: ERROR:  database "foo_test5" does not exist
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-cockroachdb-adapter-6.1.0/lib/active_record/connection_adapters/cockroachdb_adapter.rb:49:in `rescue in cockroachdb_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-cockroachdb-adapter-6.1.0/lib/active_record/connection_adapters/cockroachdb_adapter.rb:27:in `cockroachdb_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:882:in `public_send'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:882:in `new_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:926:in `checkout_new_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:905:in `try_to_checkout_new_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:866:in `acquire_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:588:in `checkout'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:428:in `connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:459:in `with_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-cockroachdb-adapter-6.1.0/lib/active_record/connection_adapters/cockroachdb_adapter.rb:66:in `initialize'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/pool_config.rb:56:in `new'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/pool_config.rb:56:in `block in pool'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/pool_config.rb:56:in `pool'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:1072:in `block in establish_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:1071:in `establish_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/connection_handling.rb:52:in `establish_connection'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/railtie.rb:222:in `block (2 levels) in <class:Railtie>'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:71:in `class_eval'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:71:in `block in execute_hook'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:61:in `with_execution_control'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:66:in `execute_hook'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:43:in `block in on_load'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:42:in `each'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:42:in `on_load'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activerecord-6.1.3.1/lib/active_record/railtie.rb:217:in `block in <class:Railtie>'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/railties-6.1.3.1/lib/rails/initializable.rb:32:in `instance_exec'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/railties-6.1.3.1/lib/rails/initializable.rb:32:in `run'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/railties-6.1.3.1/lib/rails/initializable.rb:61:in `block in run_initializers'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/railties-6.1.3.1/lib/rails/initializable.rb:60:in `run_initializers'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/railties-6.1.3.1/lib/rails/application.rb:384:in `initialize!'
/Users/pisz/work/rails/x-ion/xcrm/config/environment.rb:5:in `<main>'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.7.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.7.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.7.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.7.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/bootsnap-1.7.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:332:in `block in require'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:299:in `load_dependency'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:332:in `require'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/railties-6.1.3.1/lib/rails/application.rb:360:in `require_environment!'
/Users/pisz/.rvm/gems/ruby-2.6.3/gems/railties-6.1.3.1/lib/rails/application.rb:526:in `block in run_tasks_blocks'
```

This command should create the databases. 
With the `6.1.0.pre.beta.3` version it works.

**Comment by keithdoggett** [comment]
Looks like it may be related to this https://github.com/cockroachdb/activerecord-cockroachdb-adapter/blob/25907ece42147112cd8229866c53dba63bb0ebef/lib/active_record/connection_adapters/cockroachdb_adapter.rb#L66-L79

I'll take a look at the parallelization flow to see what's going on though. I was under the impression it would create the new database first, but it appears that's not happening.

**Comment by Piioo** [comment]
`rake parallel:setup` is still not working when databases are not crated. But this setup should create the databases.

Tested with version 6.1.1

**Comment by keithdoggett** [comment]
@Piioo I opened #207. Which should fix this.

I thought you were using the built-in ActiveRecord parallelization, which was fixed by 6.1.1, but `parallel_tests` has a slightly different flow, so it was possible to raise a `ConnectionNotEstablished` error which was previously unhandled.

**Comment by Piioo** [comment]
Its working, thanks
